test/integration/job/job_test.go

```diff
@@ -2985,3 +3004,32 @@ func updateJob(ctx context.Context, jobClient typedv1.JobInterface, jobName stri
 	})
 	return job, err
 }
+
+func addFinalizerAndDeletePods(ctx context.Context, t *testing.T, clientSet clientset.Interface, namespace string) {
+	t.Helper()
+	pods, errList := clientSet.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{})
+	if errList != nil {
+		t.Fatalf("Failed to list pods: %v", errList)
+	}
+	updatePod(t, clientSet, pods.Items, func(pod *v1.Pod) {
```

Comment:
@kannon92 do we really need a finalizer for these pods?

Reply:
@mimowo and I discussed it and my original concern was flakiness on verifying terminating pods.

I wanted to add a finalizer to slow down the test so we could actually verify terminating pods.  Otherwise, I worry that we would hit a race.